asdf/schema.py

```diff
@@ -744,7 +744,11 @@ def _validate_default(validator, default, instance, schema):
 
     resolver = _make_resolver(extension.get_default_resolver())
 
-    cls = mvalidators.create(meta_schema=meta_schema,
-                             validators=validators)
+    cls = mvalidators.create(
+        meta_schema=meta_schema,
+        validators=validators,
+        type_checker=mvalidators.Draft4Validator.TYPE_CHECKER,
+        id_of=mvalidators.Draft4Validator.ID_OF,
+    )
```

Comment:
Are these supposed to be different from the `type_checker` here

https://github.com/asdf-format/asdf/blob/4899ac53720ddced2f5f494dd52b5c3d609f6fe5/asdf/schema.py#L265-L269

and the `id_of` here (same thing)

https://github.com/asdf-format/asdf/blob/4899ac53720ddced2f5f494dd52b5c3d609f6fe5/asdf/schema.py#L270-L276

?

Reply:
And if not, a comment on why they are being redefined there but not here might be in order.  It's not clear to me.  I.e. a docstring in `_create_validator()` would help.

Reply:
This is a validator for checking a schema against its metaschema, which is a different beast from the validator that checks an ASDF file against a schema.  It isn't possible for `tuple`, `np.str_`, etc to appear in a schema and we also don't need to handle YAML tags.

Reply:
Ah, cool.  I understand now.  Thanks!